### ISSUE TITLE
SingleDatePicker doesn't fire onFocusChanged when you click outside of it

### ISSUE THREAD

**Issue by sontek** [airbnb/react-dates#271]
I believe this is supposed to fire when it gets and loses focus.

**Comment by astyfx** [comment]
@sontek this is a hotfix for that bug

```
import React from 'react'
import ReactDOM from 'react-dom'

import { SingleDatePicker } from 'react-dates'

import styles from './styles.scss'

class SingleDates extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      date: null,
      focused: false,
    }

    this.onDocumentClick = this.onDocumentClick.bind(this)
  }

  componentDidMount() {
    document.addEventListener('click', this.onDocumentClick, false)
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.onDocumentClick, false)
  }

  onDocumentClick(e) {
    if (this.state.focused && !ReactDOM.findDOMNode(this).contains(e.target)) {
      this.setState({ focused: false })
    }
  }

  render() {
    return (
      <SingleDatePicker
        ref={component => this.sdp = component}
        {...this.props}
        date={this.state.date}
        focused={this.state.focused}
        onDateChange={date => {
          this.setState({ date })
        }}
        onFocusChange={({ focused }) => {
          this.setState({ focused })
        }}
      />
    )
  }
}

export default SingleDates
```

**Comment by majapw** [comment]
This was a regression! I... accidentally removed that behavior. Whoops. This should be fixed in v6.0.1.
